addons/website_hr_recruitment/models/hr_recruitment.py

```diff
@@ -59,3 +58,71 @@ def _compute_website_url(self):
     def set_open(self):
         self.write({'website_published': False})
         return super(Job, self).set_open()
+
+
+class Country(models.Model):
+    _inherit = 'res.country'
+
+    has_job = fields.Boolean(compute='_compute_has_job', search='_search_has_job')
+
+    def _compute_has_job(self):
+        for country in self:
+            jobs = self.env['hr.job'].sudo().search(
+                [('website_published', '=', True), ('no_of_recruitment', '>', 0), ('address_id.country_id', '=', country.id)]
+            )
+            country.has_job = bool(len(jobs))
+
+    def _search_has_job(self, operator, value):
+        if operator not in ('=', '!=', '<>'):
+            raise ValueError('Invalid operator: %s' % (operator,))
+
+        if not value:
+            operator = operator == "=" and '!=' or '='
+            value = True
+
+        req = """
+            SELECT
+                p.country_id, count(*)
+            FROM
+                hr_job j
+                LEFT JOIN res_partner p
+                    ON p.id = j.address_id
+            WHERE
+                j.website_published
+                AND j.no_of_recruitment > 0
+            GROUP BY p.country_id
+            HAVING count(*) > 0
+        """
+        self.env.cr.execute(req)
+        ids = [i[0] for i in self.env.cr.fetchall()]
+        op = operator == "=" and "in" or "not in"
+        return [('id', op, ids)]
+
+
+class Department(models.Model):
+    _inherit = "hr.department"
+
+    country_id = fields.Many2one(related='company_id.country_id', store=True)
+
+    has_job = fields.Boolean(compute='_compute_has_job', search='_search_has_job')
+
+    def _compute_has_job(self):
+        for department in self:
+            jobs = self.env['hr.job'].sudo().search(
+                [('website_published', '=', True), ('no_of_recruitment', '>', 0), ('department_id', '=', department.id)]
+            )
+            department.has_job = bool(len(jobs))
+
+    def _search_has_job(self, operator, value):
+        if operator not in ('=', '!=', '<>'):
+            raise ValueError('Invalid operator: %s' % (operator,))
+
+        if not value:
+            operator = operator == "=" and '!=' or '='
+            value = True
+
+        req = "SELECT j.department_id, count(*) FROM hr_job j GROUP BY j.department_id HAVING count(*) > 0"
```

Comment:
Is the `HAVING` necessary here? It seem the `SELECT .. GROUP BY` should be enough, as all department will have jobs. And wouldn't `read_group()` do the same without manual SQL?